### ISSUE TITLE
Table Names using Underscores instead of Continuous Name

### ISSUE THREAD

**Issue by rightisleft** [typeorm/typeorm#281]
The latest alpha 3 is incorrectly changing class names into multiple underscores:

Database: Postgres
OS: Mac OS X

```typescript
@Table()
export class DTODocument {
  @PrimaryGeneratedColumn() id: number;
}
```

Expected - dto_document OR DTODocument
Actual -  public | d_t_o_document       | table | rightisleft

**Comment by pleerock** [comment]
Looks like a small bug. You are able to fix it [there](https://github.com/typeorm/typeorm/blob/master/src/util/StringUtils.ts) ?

**Comment by valeriy-devpronet** [comment]
what if use [lodash.snakeCase](https://github.com/lodash/lodash/blob/4.17.4/lodash.js#L14542)  or [lodash.camelCase](https://github.com/lodash/lodash/blob/4.17.4/lodash.js#L14117) or this is unexpected dependecies?

**Comment by pleerock** [comment]
no, typeorm should not have external dependecies like this because its also packed for browser where user will require to install lodash separately to make it to work. Thats why we need our own function.

**Comment by pleerock** [comment]
thanks to @mrkmg this is fixed and will be released in next 0.2.0-alpha version (`npm i typeorm@next`)

**Comment by hi2u** [comment]
@rightisleft when you mentioned...

> Expected - dto_document OR DTODocument

Do you know if there's some way to switch TypeORM between using camelCase and under_scores ?

I'm trying to figure out how to use underscores in foreign key column names: https://github.com/typeorm/typeorm/issues/2608 ... not sure if my issue is related.
